AWS/aws-cloudtrail/ingest/parser.yml

```
@@ -0,0 +1,111 @@
+          event.dataset: 'cloudtrail'
+          event.action: '{{json_event.message.eventName or json_event.message.insightDetails.eventName}}'
+          event.code: '{{json_event.message.errorCode or json_event.message.insightDetails.errorCode}}'
+          event.reason: '{{json_event.message.errorMessage}}'
+          event.provider: '{{json_event.message.eventSource or json_event.message.insightDetails.eventSource}}'
+          cloud.provider: 'aws'
+          cloud.service.name: 'cloudtrail'
+          cloud.region: '{{json_event.message.awsRegion}}'
+          cloud.account.id: '{{json_event.message.userIdentity.accountId}}'
+          cloud.instance.id: '{{json_event.message.requestParameters.instanceId}}'
+          action.type: '{{json_event.message.eventType}}'
+          action.name: '{{json_event.message.eventName}}'
+          action.outcome: 'success'
+          action.target: 'network-traffic'
+          user_agent.original: '{{json_event.message.userAgent}}'
+          user.id: '{{json_event.message.userIdentity.accountId}}'
+      - set:
+          source.address: '{{json_event.message.sourceIPAddress}}'
+          source.ip: '{{parsed_source.source.ip}}'
+          source.domain: '{{parsed_source.source.domain}}'
+        filter: '{{json_event.message.sourceIPAddress != "AWS Internal"}}'
+      - set:
+          user.name: '{{json_event.message.responseElements.user.userId}}'
+        filter: '{{json_event.message.get("responseElements") != None}}'
+      - set:
+          tls.cipher: '{{json_event.message.tlsDetails.cipherSuite}}'
+          tls.version: '{{json_event.message.tlsDetails.tlsVersion}}'
+        filter: '{{json_event.message.get("tlsDetails") != None}}'
+      - set:
+          event.category: ['iam']
+          event.type: ['change']
+        filter: '{{json_event.message.eventName in ("PasswordUpdated", "EnableMFADevice")}}'
+      - set:
+          event.category: ['authentication']
+        filter: '{{json_event.message.eventName == "ConsoleLogin"}}'
+      - translate:
+        dictionary:
+          Success: ['allowed']
+          Failure: ['denied']
+        mapping:
+          json_event.message.responseElements.ConsoleLogin: event.type
+        filter: '{{json_event.message.eventName == "ConsoleLogin"}}'
+
+  set_action_fields:
+    actions:
+      - set:
+          action.properties.errorCode: '{{json_event.message.errorCode}}'
+          action.properties.errorMessage: '{{json_event.message.errorMessage}}'
+          action.properties.recipientAccountId: '{{json_event.message.recipientAccountId}}'
+          action.properties.userIdentity: '{{json_event.message.userIdentity}}'
+          action.properties.resources: '{{json_event.message.resources}}'
+          action.properties.requestParameters.userData: '{{json_event.message.requestParameters.userData}}'
+          action.properties.responseElements.publiclyAccessible: '{{json_event.message.responseElements.publiclyAccessible}}'
```

(suggestion) Maybe, you could have written:
```suggestion
          action.properties: '{{json_event.message}}'
```

---

We have to control the fields produced by the parser. I prefer to specify each field to copy

---

FYI if `action.properties` is declared as a `dict` field, the produced fields are controlled. Only `object` fields are bypassing validation.